peer.go

```diff
@@ -221,23 +227,29 @@ func (l *PeerList) exists(hostPort string) (*peerScore, uint64, bool) {
 	return ps, score, ok
 }
 
-// updatePeer is called when there is a change that may cause the peer's score to change.
+// onPeerChange is called when there is a change that may cause the peer's score to change.
 // The new score is calculated, and the peer heap is updated with the new score if the score changes.
-func (l *PeerList) updatePeer(p *Peer) {
+func (l *PeerList) onPeerChange(p *Peer) {
 	ps, psScore, ok := l.exists(p.hostPort)
 	if !ok {
 		return
 	}
 
-	newScore := l.scoreCalculator.GetScore(p)
```

Comment:
This makes the behaviour of `onPeerChange` always get a write lock on the peer list, even if the score doesn't change. I don't think we want this behaviour.

Can we revert back to calculating the score, and only doing `updatePeer` if the score has changed.

Within `updatePeer`, it's worth checking the score as well (Since it's possible it changed between the read lock and the write lock).